### PR TITLE
Add internal api docs

### PR DESCRIPTION
**Pull request by Gerg**
First attempt, probably not 100% complete or accurate

### CODE REVIEW

File: docs/internal/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+**Intended Consumer:** Diego staging task
+
+**Auth Mechanism:** Basic Auth
+
+### GET /internal/v2/droplets/:guid/:droplet_checksum/download
+**Description:** (Deprecated) Permalink for downloading a droplet. Use `GET /internal/v4/droplets/:guid/:droplet_checksum/download` instead.
+
+**Intended Consumer:** Diego running task or process
+
+**Auth Mechanism:** None
```
Comment (tcdowney) [inline docs/internal/README.md]: Can you note that this endpoint just redirects to the `/internal/v4` mTLS authenticated one. It doesn't let you actually download the droplet without any auth.

File: docs/internal/README.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+**Description:** Marks package as uploaded
+
+**Intended Consumer:** Bits Service
+
+**Auth Mechanism:** MTLS
+
+### GET /internal/v4/syslog_drain_urls
+**Description:** Return list of syslog drain urls from logging services
+
+**Intended Consumer:** Loggregator
```
Comment (tcdowney) [inline docs/internal/README.md]: Do we know what piece (these days at least) of the Loggregator subsystem actually hits this?

File: docs/internal/README.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+**Auth Mechanism:** MTLS
+
+### POST /internal/v4/tasks/:task_guid/completed
+**Description:** Marks task as complete
+
+**Intended Consumer:** Diego running task
```
Comment (tcdowney) [inline docs/internal/README.md]: Maybe clarify that this callback is for CF app tasks that run on Diego

https://v3-apidocs.cloudfoundry.org/version/3.71.0/index.html#tasks

File: docs/internal/README.md (hunk 2 above)
Comment (tcdowney) [reply docs/internal/README.md]: I'm finding out here:
https://cloudfoundry.slack.com/archives/C02HCCXV5/p1558560940076900?thread_ts=1558560940.076900&cid=C02HCCXV5

File: docs/internal/README.md (hunk 1 above)
Comment (Gerg) [reply docs/internal/README.md]: I think there is still technically aslo an unauthenticated path: https://github.com/cloudfoundry/cloud_controller_ng/blob/3a93d5fea615fa085b19c2c940c21ad0dfb05948/app/controllers/internal/download_droplets_controller.rb#L24

**Comment by tcdowney** [comment]
Looks great! Thanks @Gerg. 😊 👍 👍

**Pull request merged**